pkg/test/framework/components/pilot/pilot.go

```diff
@@ -33,14 +29,6 @@ type TypeURL string
 // Instance of Pilot
 type Instance interface {
 	resource.Resource
-
-	CallDiscovery(req *discovery.DiscoveryRequest) (*discovery.DiscoveryResponse, error)
```

Comment:
@stevenctl not sure if this impacts your work at all to merge the components?

Reply:
Looks like it makes the pilot bit unnecessary :)

Reply:
Yeah actually maybe we can just delete Pilot now instead of merge it. Need to check if it serves any purpose after this PR

Reply:
+1